### PR TITLE
refactor: move from io/ioutil to io and os packages

### PR DESCRIPTION
**Pull request by mmorel-35**
In go 1.16 io/ioutil package is deprecated, see https://go.dev/doc/go1.16#ioutil

### CODE REVIEW

**Review by metalmatze — COMMENTED** [review]
Very nice, no surprises in the code! 👍 
There seems to be either a flaky test or something slightly off with moving to the other/new Temp funcs:
https://app.circleci.com/pipelines/github/prometheus/prometheus/16508/workflows/301b1fdb-1683-4997-9bdd-e6fbc7d15565/jobs/79014?invite=true#step-103-18

File: tsdb/db.go [hunk 1]
```diff
@@ -761,20 +760,24 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
...
 	if os.IsNotExist(err) {
 		return nil
 	}
 	if err != nil {
 		return err
 	}
-	for _, fi := range files {
+	for _, f := range files {
+		fi, err := f.Info()
+		if err != nil {
```
Comment (mmorel-35) [inline tsdb/db.go:772]: Shall this error just be ignored then ?

Comment (metalmatze) [reply tsdb/db.go:772]: I don't think it should be ignore. After all the next call depends on `fi` being there. Is there a problem with returning the error here?

Comment (mmorel-35) [reply tsdb/db.go]: The call to Ìnfo()`can be avoid by using directly `fs.DirEntry` instead of `os.FileInfo`. Let's see if it fixes the tests

**Review by cstyan — APPROVED** [review]

**Review by metalmatze — APPROVED** [review]

**Comment by mmorel-35** [comment]
Hi! What is blocking the merge? Are you expecting me to do anything else?

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**